pkg/apis/machine/validation/alicloudmachineclass.go

```diff
@@ -63,6 +66,43 @@ func validateAlicloudMachineClassSpec(spec *machine.AlicloudMachineClassSpec, fl
 		allErrs = append(allErrs, field.Required(fldPath.Child("keyPairName"), "KeyPairName is required"))
 	}
 
+	const dataDiskNameFmt string = `[a-zA-Z0-9\.\-_:]+`
```

Comment:
it should starts with a letter. The regexp looks like `[a-zA-Z][a-zA-Z0-9\.\-_:]+`